### ISSUE TITLE
[iOS] Flakey race condition between SetSemanticsTreeEnabled and UpdateSemantics

### ISSUE THREAD

**Issue by vashworth** [flutter/flutter#174842]
### Steps to reproduce

1. `flutter create my_app`
2. `cd my_app`
3. `et run --config ios_debug_sim_unopt_arm64`
4. Quit the app once it launches
5. Open `my_app/ios/Runner.xcworkspace`
6. In Xcode do Product > Run 

Note: Error does not always happen since it's a race condition, you may need to do step 6 multiple times. It is more likely to happen if you [setup UIScene in your Info.plist](https://docs.flutter.dev/release/breaking-changes/uiscenedelegate#info-plist-migration), but I have seen it happen for both. Also, the crash can also happen when running from the Flutter CLI, but running from Xcode is faster and allows you to see the error log.

### Expected results

App should run fine when using a local engine

### Actual results

App crashes

```
[FATAL:flutter/shell/platform/darwin/ios/platform_view_ios.mm(156)] Check failed: accessibility_bridge_.
```

https://github.com/flutter/flutter/blob/7a13e1be3a98704f91acf2807aede12277771bf6/engine/src/flutter/shell/platform/darwin/ios/platform_view_ios.mm#L154

I tried tracing it down and it appears that occasionally `FlutterEngine` enables semantics before the `_handleFrameworkSemanticsEnabledChanged` listener is set up.

https://github.com/flutter/flutter/blob/7a13e1be3a98704f91acf2807aede12277771bf6/engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterEngine.mm#L345-L351

https://github.com/flutter/flutter/blob/7a13e1be3a98704f91acf2807aede12277771bf6/packages/flutter/lib/src/semantics/binding.dart#L42

And therefore, the semantics tree is never enabled:
https://github.com/flutter/flutter/blob/7a13e1be3a98704f91acf2807aede12277771bf6/packages/flutter/lib/src/semantics/binding.dart#L168-L170

Likely caused by https://github.com/flutter/flutter/pull/174163.

### Code sample

Use default `flutter create` sample app

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel main, 3.36.0-1.0.pre-348, on macOS 15.6.1 24G90 darwin-arm64, locale en)
[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 16.3)
[✓] Chrome - develop for the web
[✓] Connected device (4 available)
[✓] Network resources

• No issues found!
```

</details>

**Comment by vashworth** [comment]
I think this is likely also impacting non-local engine apps, but that's the use case I discovered it with.

cc @chunhtai Do you think this warrants a revert of https://github.com/flutter/flutter/pull/174163?

**Comment by chunhtai** [comment]
Let me take a look, this seems like a bug that surface by the change. If i can't figure out something quick, I will revert

**Comment by chunhtai** [comment]
actually this seems like a corner case in the pr https://github.com/flutter/flutter/pull/174163, but should be a easy fix. I will attempt to fix forward due to the initial pr requires g3fix. Let me know otherwise

**Comment by chunhtai** [comment]
mark it as P0 since this is serious
